tss-esapi/tests/integration_tests/structures_tests/tagged_tests/tagged_signature_scheme_tests.rs

```diff
@@ -4,200 +4,215 @@
 use tss_esapi::{
     interface_types::algorithm::{HashingAlgorithm, SignatureSchemeAlgorithm},
     structures::{EcDaaScheme, HashScheme, HmacScheme, SignatureScheme},
-    tss2_esys::{
-        TPMS_SCHEME_ECDAA, TPMS_SCHEME_HASH, TPMS_SCHEME_HMAC, TPMT_SIG_SCHEME, TPMU_SIG_SCHEME,
-    },
+    tss2_esys::{TPMT_SIG_SCHEME, TPMU_SIG_SCHEME},
     Error, WrapperErrorKind,
 };
 
-use std::convert::{TryFrom, TryInto};
+use std::convert::TryFrom;
 
-fn validate_tss_hash_scheme(
-    left: &TPMS_SCHEME_HASH,
-    right: &TPMS_SCHEME_HASH,
-    union_field_name: &str,
-) {
-    assert_eq!(
-        left.hashAlg, right.hashAlg,
-        "{} in details, hashAlg did not match",
-        union_field_name
-    );
+macro_rules! assert_signing_scheme {
+    (SignatureScheme::$scheme_item:ident, $scheme_type:ty, HashingAlgorithm::$hash_item:ident, $count_expr:expr) => {
+        assert_signing_scheme!(
+            SignatureScheme::$scheme_item{scheme: <$scheme_type>::new(HashingAlgorithm::$hash_item, $count_expr)},
+            HashingAlgorithm::$hash_item
+        );
+    };
+    (SignatureScheme::$scheme_item:ident, $scheme_type:ty, HashingAlgorithm::$hash_item:ident) => {
+        assert_signing_scheme!(
+            SignatureScheme::$scheme_item{scheme: <$scheme_type>::new(HashingAlgorithm::$hash_item)},
+            HashingAlgorithm::$hash_item
+        );
+    };
+    (SignatureScheme::$scheme_item:ident, HashingAlgorithm::$hash_item:ident) => {
+        assert_signing_scheme!(
+            SignatureScheme::$scheme_item{scheme: HashScheme::new(HashingAlgorithm::$hash_item)},
+            HashingAlgorithm::$hash_item
+        );
+    };
+    ($item_expr:expr, HashingAlgorithm::$hash_item:ident) => {
+        let item = $item_expr;
+        let actual = item.signing_scheme().unwrap_or_else(|_| {
+            panic!(
+                "A valid {} that contains a scheme should not return an error when calling the `signing_scheme`method.",
```

Comment:
```suggestion
                "A valid {} that contains a scheme should not return an error when calling the `signing_scheme` method.",
```

Reply:
Fixed that and the same problem in other places.